ui/src/app/cohort-search/tree-node/tree-node.component.tsx

```diff
@@ -292,7 +292,9 @@ export class TreeNode extends React.Component<TreeNodeProps, TreeNodeState> {
       setAttributes} = this.props;
     const {children, error, expanded, hover, loading, searchMatch} = this.state;
     const nodeChildren = domainId === DomainType.PHYSICALMEASUREMENT.toString() ? node.children : children;
-    const selected = selectedIds.includes(this.paramId) || groupSelections.includes(parentId);
+    const selected = serverConfigStore.getValue().enableCohortBuilderV2
+      ? currentCohortCriteriaStore.getValue().some(crit => crit.parameterId === this.paramId)
+      : selectedIds.includes(this.paramId) || groupSelections.includes(parentId);
```

Comment:
Nit: can you please add brackets, because of so many conditions its a little difficult to read